crates/util/python-utils/src/bridge/types.rs

```diff
@@ -0,0 +1,166 @@
+//! Bridge transaction data structures
+//!
+//! This module contains the core data structures for bridge operations,
+//! adapted from the mock-bridge implementation for use in python-utils.
+
+use bdk_wallet::bitcoin::{
+    consensus, script::PushBytesBuf, Amount, OutPoint, ScriptBuf, TapNodeHash, Txid, XOnlyPublicKey,
+};
+use pyo3::prelude::*;
+use secp256k1::Keypair;
+
+use crate::{taproot::musig_aggregate_pks_inner, utils::parse_operator_keys};
+
+#[pyclass]
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct DepositRequestData {
+    /// The deposit request transaction outpoints from the users.
+    pub deposit_request_outpoint: OutPoint,
+
+    /// The stake index that will be tied to this deposit.
+    ///
+    /// This is required in order to make sure that the at withdrawal time, deposit UTXOs are
+    /// assigned in the same order that the stake transactions were linked during setup time
+    ///
+    /// # Note
+    ///
+    /// The stake index must be encoded in 4-byte big-endian.
+    pub stake_index: u32,
+
+    /// The execution environment address to mint the equivalent tokens to.
+    /// As of now, this is just the 20-byte EVM address.
+    pub ee_address: Vec<u8>,
+
+    /// The amount in bitcoins that the user is sending.
+    ///
+    /// This amount should be greater than the bridge denomination for the deposit to be
+    /// confirmed on bitcoin. The excess amount is used as miner fees for the Deposit Transaction.
+    pub total_amount: Amount,
+
+    /// The [`XOnlyPublicKey`] in the Deposit Request Transaction (DRT) as provided by the
+    /// user in their `OP_RETURN` output.
+    pub x_only_public_key: XOnlyPublicKey,
+
+    /// The original script_pubkey in the Deposit Request Transaction (DRT) output used to sanity
+    /// check computation internally i.e., whether the known information (n/n script spend path,
+    /// [`static@UNSPENDABLE_INTERNAL_KEY`]) + the [`Self::take_back_leaf_hash`] yields the same
+    /// P2TR address.
+    pub original_script_pubkey: ScriptBuf,
+}
+
+#[pymethods]
+impl DepositRequestData {
+    #[getter]
+    fn stake_index(&self) -> u32 {
+        self.stake_index
+    }
+
+    #[getter]
+    fn ee_address(&self) -> Vec<u8> {
+        self.ee_address.clone()
+    }
+
+    #[getter]
+    fn total_amount(&self) -> u64 {
+        self.total_amount.to_sat()
+    }
+
+    #[getter]
+    fn x_only_public_key(&self) -> String {
+        self.x_only_public_key.to_string()
+    }
```

Comment:
Are these getters necessary if the fields are pub?  If yes, then the fields should not be pub.